### ISSUE TITLE
exit 0 when connection failed

### ISSUE THREAD

**Issue by Tamaranch** [ProtonVPN/protonvpn-cli#114]
Hi,

Is it wanted that protonvpn-cli exits with status 0 when connection failed but quiet disconnection succeeded? Shouldn't openvpn_disconnect() being called with dont_exit argument just before exiting with status 1 in openvpn_connect()?

Best regards.

**Comment by mazen160** [comment]
Hi @Tamaranch 

This is used to do further actions. on https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L444

It's only used when uninstalling protonvpn-cli, so that if 
```
  if [[ "$1" != "quiet" ]]; then
    echo "[!] Error disconnecting OpenVPN."

    if [[ "$2" != "dont_exit" ]]; then
      exit 1
    fi

  fi
```

If it failed on uninstallation, the installation will not continue, and return an exit status code of 1
https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L589

I had reviewed it now:)

Thanks,
Mazin

**Comment by Tamaranch** [comment]
Just in case, I precise my previous post, which was not very well written: In my case, for some reason, protonvpn-cli often fails to connect at its first attempt, giving me "[[!] Error connecting to VPN](https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L520)" on stdout, but returning an exit status code of 0.
I realized that by trying to use a statement of the form `while ! protonvpn-cli -f; do …; done` in a script, that I had to replace by `while ! protonvpn-cli -f | grep -q "Connected!"; do …; done`.
The "problem" (if it is one) that I mentioned in my previous post, is that, in this case, `openvpn_connect()` calls `openvpn_disconnect()` at [line 524](https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L524) with only one argument (quiet), making the whole program exit with an exit status code of 0 at [line 440](https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L440), instead of calling `openvpn_disconnect()` with two arguments (quiet, dont_exit), which would make the whole program exit at [line 525](https://github.com/ProtonVPN/protonvpn-cli/blob/master/protonvpn-cli.sh#L525) with an exit status code of 1 (which is the behavior I expected).
I hope it is clearer like that, and anyway, I apologize for my bad English and bad writing!

Best regards.
